src/Psalm/Internal/Analyzer/Statements/Expression/Call/StaticMethod/AtomicStaticCallAnalyzer.php

```diff
@@ -1002,4 +1009,41 @@ public static function handleNonObjectCall(
             $statements_analyzer->getSuppressedIssues()
         );
     }
+
+    /**
+     * Try to find matching pseudo method over ancestors (including interfaces).
+     *
+     * Returns the pseudo method if exists, with its defining class storage.
+     * If the method is not declared, null is returned.
+     *
+     * @param Codebase $codebase
+     * @param ClassLikeStorage $static_class_storage The called class
+     * @param lowercase-string $method_name_lc
+     *
+     * @return array{MethodStorage, ClassLikeStorage}|null
+     */
+    private static function findPseudoMethodAndClassStorages(
+        Codebase $codebase,
+        ClassLikeStorage $static_class_storage,
+        string $method_name_lc
+    ): ?array {
+        if ($pseudo_method_storage = $static_class_storage->pseudo_static_methods[$method_name_lc] ?? null) {
+            return [$pseudo_method_storage, $static_class_storage];
+        }
+
+        $ancestors = $static_class_storage->class_implements + $static_class_storage->parent_classes;
```

Comment:
I'm still not sure what parent_classes really contains, do you?

I'm also surprised because as far as I can tell, inheritance with parent classes worked (as opposed to interfaces): https://psalm.dev/r/be4d8b2941

So I'm not sure if the parent_classes part is needed

Reply:
I've to admit it, I don't know what parent_classes exactly contains 😅 
It seems to contains all parents hierarchy (cf: https://github.com/vimeo/psalm/blob/36d5a2a83c1d7b3e2aca1cfe9e9dcc916619e962/src/Psalm/Internal/Codebase/Reflection.php#L104)

It's really weird, because inheritance dot not works for static call. 
Removing `parent_classes` from `MissingMethodCallHandler::findPseudoMethodAndClassStorages()` will works visibly (all tests pass), but it's required for `AtomicStaticCallAnalyzer::findPseudoMethodAndClassStorages()` (tests fails without).